starboard/nplb/vertical_video_test.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "starboard/media.h"
+#include "starboard/nplb/player_test_fixture.h"
+#include "starboard/nplb/player_test_util.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+using ::testing::ValuesIn;
+
+typedef SbPlayerTestFixture::GroupedSamples GroupedSamples;
+
+class VerticalVideoTest : public ::testing::TestWithParam<SbPlayerTestConfig> {
+};
+
+TEST(VerticalVideoTest, CapabilityQuery) {
+  auto check_vertical_resolution_support =
+      [](SbMediaSupportType support, std::string width, std::string height) {
+        if (support != kSbMediaSupportTypeProbably) {
+          SB_LOG(INFO) << "Vertical resolution " << width << "x" << height
+                       << " should be supported on this platform.";
+        }
+      };
+
+  SbMediaSupportType support = SbMediaCanPlayMimeAndKeySystem(
+      "video/mp4; codecs=\"avc1.4d002a\"; width=1920; height=1080", "");
+  if (support == kSbMediaSupportTypeProbably) {
+    support = SbMediaCanPlayMimeAndKeySystem(
+        "video/mp4; codecs=\"avc1.4d002a\"; width=608; height=1080", "");
+    check_vertical_resolution_support(support, "608", "1080");
```

Comment:
As the vertical resolution requirement is not yet enforced, we don't need to fail if a given resolution is unsupported. I may update the log as technically the resolution doesn't have to be supported.